pkg/deploy/kubernetes_image_puller.go

```diff
@@ -379,6 +385,88 @@ func UpdateImagePullerSpecIfEmpty(ctx *DeployContext) (orgv1.CheClusterSpecImage
 	return ctx.CheCluster.Spec.ImagePuller, nil
 }
 
+func AddRelatedImages(ctx *DeployContext) {
+	relatedImages := GetRelatedImagesEnvVars()
+
+	if len(relatedImages) == 0 {
+		return
+	}
+
+	if len(ctx.CheCluster.Spec.ImagePuller.Spec.Images) > 0 &&
+		!strings.HasSuffix(ctx.CheCluster.Spec.ImagePuller.Spec.Images, ";") {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += ";"
+	}
+
+	for _, envVar := range relatedImages {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += envVar.Name + "=" + envVar.Value + ";"
+	}
+	ctx.ClusterAPI.Client.Update(context.TODO(), ctx.CheCluster, &client.UpdateOptions{})
```

Comment:
pls use https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/checluster.go#L24 and handle error

Reply:
thank you, fixed